### PR TITLE
License calendar event

### PR DESCRIPTION
**Pull request by FarazA22**
### Ticket

https://www.pivotaltracker.com/story/show/188079116

### Approach
There are two pre-factor commits associated with this work. The first pre-factor involved changing the URL path of license calendar events, and the 2nd pre-factor involved renaming the licenses folder to help clarify to developers the intent of the files within the folder. I did this work because I felt the existing logic was confusing, I think the changes that I've made made the code base a little better.

Regarding the feature work, I had to update both anytime action reinstatement files and the license calendar events files that resided within the content folder. These changes were discussed with content during standup.

I refactored the logic that displays the permit status into its own component since it is being used in two places. I also re-factored some of the tests that were impacted due to the updates, for example, that new component had its own test that tested all of the rendering logic.

Lastly, the management search and CMS was a little funky so I made some changes there to fix some of the search capability and previews within the CMS

### Steps to Test
To review anytime action reinstatement changes, you need to search for a license that is expired. To view a calendar event you need to have a license that expires in the future.

## Code author checklist

- [x] I have rebased this branch from the latest main branch
- [x] I have performed a self-review of my code
- [x] I have created and/or updated relevant documentation on the engineering documentation website
- [x] I have not used any relative imports
- [x] I have pruned any instances of unused code
- [x] I have not added any markdown to labels, titles and button text in config
- [x] If I added/updated any values in `userData` (including `profileData`, `formationData` etc), then I added a new migration file
- [x] I have checked for and removed instances of unused config from CMS
- [x] If I added any new collections to the CMS config, then I updated the search tool and `cmsCollections.ts` (see CMS Additions in Engineering Reference/FAQ on the engineering documentation site)
- [x] I have updated relevant `.env` values in both `.env-template` and in Bitwarden

### CODE REVIEW

File: web/src/components/dashboard/AnytimeActionDropdown.test.tsx [hunk 1]
```diff
@@ -220,7 +220,7 @@ describe("<AnytimeActionDropdown />", () => {
       expect(task.compareDocumentPosition(licenseLast)).toBe(Node.DOCUMENT_POSITION_FOLLOWING);
     });
 
-    it("filters out licenses unrelated to user industry", () => {
```
Comment (FarazA22) [inline web/src/components/dashboard/AnytimeActionDropdown.test.tsx:223]: revert - this is an error

File: web/src/components/tasks/LicenseCurrentStatusComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+          <ChecklistTag status={item.status} />
+          <span className="tablet:margin-left-2 text-left width-full">{item.title}</span>
+        </div>
+      </div>
+    );
+  };
+
+  return (
+    <div data-testid="license-current-status-component">
+      <div className="h3-styling">{Config.currentLicenseStatusDefaults.licenseStatusHeaderText}</div>
```
Comment (adidner) [inline web/src/components/tasks/LicenseCurrentStatusComponent.tsx:54]: pebble - should this actually be an H3? or at least a heading tag of some kind?

File: web/src/components/tasks/LicenseCurrentStatusComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+      <div className="drop-shadow-xs padding-3 radius-lg">
+        {licenseDetails?.licenseStatus && (
+          <LicenseStatusHeader licenseStatus={licenseDetails.licenseStatus} />
+        )}
+        <HorizontalLine />
+        <Accordion onChange={() => {}}>
+          <AccordionSummary
+            aria-controls=""
+            expandIcon={<Icon className={"usa-icon--size-3 text-base-light"}>expand_more</Icon>}
+            className={"h4-styling"}
```
Comment (adidner) [inline web/src/components/tasks/LicenseCurrentStatusComponent.tsx:65]: pebble - Same with this, should this be a heading tag? I feel like a lot of our AccordianSummary component contain heading tags

**Review by adidner — APPROVED** [review]
Generally code looks great, a few minor questions and curiosities

File: web/public/mgmt/config.yml [hunk 4]
```diff
@@ -3664,6 +3653,40 @@ collections:
     folder: "content/src"
     fields:
       - { name: "" }
+  - name: "current-license-status-defaults"
+    label: "Current License Status Defaults"
```
Comment (adidner) [inline web/public/mgmt/config.yml:3657]: dust - personallly not a huge fan of current as a name for variables typically.

File: web/src/components/tasks/license-calendar-event/LicenseCalendarEventElement.tsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+          <h1>
+            {props.licenseEventType === "renewal"
+              ? props.license.renewalEventDisplayName
+              : props.license.expirationEventDisplayName}
+          </h1>
+          <div>
+            <span className="text-bold">
+              {" "}
+              {Config.currentLicenseStatusDefaults.licenseExpirationHeaderText}
+            </span>{" "}
```
Comment (adidner) [inline web/src/components/tasks/license-calendar-event/LicenseCalendarEventElement.tsx:50]: dust - not sure why we have random {" "} stuff going on here, care to clarify for me?

File: web/public/mgmt/config.yml [hunk 6]
```diff
@@ -3115,16 +3115,23 @@ collections:
         }
       - { label: "Renewal Event Label", name: "renewalEventDisplayName", widget: "string", required: true }
       - {
-          label: "Above License Status Content",
-          name: "aboveLicenseStatusContent",
+          label: "Disclaimer Text",
```
[the remark this answers is not in the document]
Comment (SirSamuelJoseph) [reply web/public/mgmt/config.yml:3118]: Definitely agree that position is changing whereas concepts stay clear, great refactor Faraz!

File: web/src/components/tasks/license-calendar-event/LicenseCalendarEventElement.tsx [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+
+interface LicenseElementProps {
+  license: LicenseEventType;
+  licenseEventType: LicenseEventSubtype;
+  licenseName: string;
+  CMS_ONLY_fakeBusiness?: Business;
+  CMS_ONLY_fakeLicenseName?: LicenseName;
+}
+
+export const LicenseElement = (props: LicenseElementProps): ReactElement => {
```
Comment (FarazA22) [inline web/src/components/tasks/license-calendar-event/LicenseCalendarEventElement.tsx:23]: Update the name to match filename

File: web/src/lib/static/loadLicenses.ts [hunk 8]
```diff
@@ -9,7 +9,7 @@ export type LicenseUrlSlugParam = {
   licenseUrlSlug: string;
 };
 
-const licensesDir = path.join(process.cwd(), "..", "content", "src", "licenses");
+const licensesDir = path.join(process.cwd(), "..", "content", "src", "license-calendar-events");
```
[the remark this answers is not in the document]
Comment (FarazA22) [reply web/src/lib/static/loadLicenses.ts:13]: Thanks for catching this... Love the idea, and will implement it as part of the work

File: web/public/mgmt/config.yml (hunk 4 above)
Comment (FarazA22) [reply web/public/mgmt/config.yml:3657]: changed

File: web/src/components/tasks/LicenseCurrentStatusComponent.tsx (hunk 2 above)
Comment (FarazA22) [reply web/src/components/tasks/LicenseCurrentStatusComponent.tsx:54]: added h2 heading with h3 variant

File: web/src/components/tasks/LicenseCurrentStatusComponent.tsx (hunk 3 above)
Comment (FarazA22) [reply web/src/components/tasks/LicenseCurrentStatusComponent.tsx:65]: added h3 with h4 variant

File: web/src/components/tasks/license-calendar-event/LicenseCalendarEventElement.tsx (hunk 5 above)
Comment (FarazA22) [reply web/src/components/tasks/license-calendar-event/LicenseCalendarEventElement.tsx:50]: I delete the space on line 48, it's not needed, but the {" "} creates a space between the span and the text that immediately follows it.

Comment (FarazA22) [reply web/src/components/tasks/license-calendar-event/LicenseCalendarEventElement.tsx:50]: It's another way of adding a space

**Pull request merged**